samples/snippets/src/main/java/CustomCredentialSupplierOktaWorkload.java

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import com.google.api.client.json.GenericJson;
+import com.google.api.client.json.gson.GsonFactory;
+import com.google.auth.oauth2.ExternalAccountSupplierContext;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.auth.oauth2.IdentityPoolCredentials;
+import com.google.auth.oauth2.IdentityPoolSubjectTokenSupplier;
+import com.google.cloud.storage.Bucket;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+import java.io.BufferedReader;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+
+/**
+ * This sample demonstrates how to use a custom subject token supplier to authenticate to Google
+ * Cloud Storage, using Okta as the identity provider.
+ */
+public class CustomCredentialSupplierOktaWorkload {
+
+  public static void main(String[] args) {
+    // TODO(Developer): Replace these variables with your actual values.
+    String gcpWorkloadAudience = System.getenv("GCP_WORKLOAD_AUDIENCE");
+    String serviceAccountImpersonationUrl =
+        System.getenv("GCP_SERVICE_ACCOUNT_IMPERSONATION_URL");
+    String gcsBucketName = System.getenv("GCS_BUCKET_NAME");
+    String oktaDomain = System.getenv("OKTA_DOMAIN");
+    String oktaClientId = System.getenv("OKTA_CLIENT_ID");
+    String oktaClientSecret = System.getenv("OKTA_CLIENT_SECRET");
```

Comment:
Same here, do you have any docs on how a user can find the okta domain, client_id, secret? Probably pretty straightforward, but could be helpful to people.

Reply:
Ah, I just saw the update. Resolving this.